tests/sentry/models/test_user.py

```diff
@@ -122,6 +125,23 @@ def test_cascades_to_regions_even_if_user_ownership_revoked(self):
             schedule_hybrid_cloud_foreign_key_jobs()
         assert self.get_user_saved_search_count() == 0
 
+    def test_update_purge_region_cache(self):
+        user = self.create_user()
+        na_org = self.create_organization(region=_TEST_REGIONS[0])
+        self.create_member(user=user, organization=na_org)
+
+        with patch.object(caching_module, "region_caching_service") as mock_caching_service:
+            user.username = "bob2"
+            user.save()
+            mock_caching_service.clear_key.assert_any_call(
+                key=f"user_service.get_many_by_id:{user.id}",
+                region_name=_TEST_REGIONS[0].name,
+            )
+            mock_caching_service.clear_key.assert_any_call(
+                key=f"user_service.get_user:{user.id}",
+                region_name=_TEST_REGIONS[0].name,
+            )
```

Comment:
Is this along the lines of what you were looking for @GabeVillalobos (from a [previous comment](https://github.com/getsentry/sentry/pull/71719/files#r1624797976)) ?